packages/cascara/src/ui/Form/Form.js

```diff
@@ -101,18 +132,20 @@ const formFields = (display, data) => {
 };
 
 const renderActions = (actions) =>
-  actions.map((action, id) => {
+  actions?.modules?.map((action, idx) => {
     const { module, ...rest } = action;
-    const Action = bundledActionModules[module];
+    const Action = availableActionModules[module];
 
-    //
     // In certain predefined-action modules in which a label is not required, e.g. `edit`,
-    // the following unique key generation fails, as it relies on the label (content).
-    const key = `${id}.${module}.${rest.label || module}`;
+    // the following unique key generation fails, as it relies on the label.
+    const key = `${idx}.${module}.${rest.label || module}`;
 
     // TODO: create a function to filter out props that we don't want
+    // eslint-disable-next-line react/forbid-foreign-prop-types -- @brian we need to see if this approach is what we want
+    const { myProps } = filterProps(rest, Action.propTypes);
+
```

Comment:
@brianespinosa is [this](https://github.com/Espressive/cascara/pull/225/files#diff-722b073477d424a954017a1cc1c84b8904435755f9099bc392e810383b9eaf7cR63) `filterProps` function what we want?

Perhaps change the shape returned, with a more semantic name..

Reply:
Kind of. I think we can rely on prior art here. Take a look at what is happening in the code in libraries like Semantic UI React and also Reakit. There are helper functions to evaluate these slices of props. I think the functions are semantically named something like `handledProps` which makes sense in the context of the component that the library is handling them.

Reply:
@lordpixel go ahead and make a separate task to do this research and settle on a pattern. I think that does not need to be part if the early Q3 release so it could go into the Table P3 epic.

This code can stay since it works for now and we can come back to evaluate this when we look at the overall possible hooks based useTableState style API.